shopfloor_reception/services/reception.py

```diff
@@ -1237,15 +1258,34 @@ def _post_shopfloor_created_line(self, selected_line):
         )
 
     def _auto_post_line(self, selected_line):
-        new_move = selected_line.move_id.split_other_move_lines(
-            selected_line, intersection=True
+        # If user only processed 1/5 and is the only one working on the move,
+        # then selected_line is the only one related to this move.
+        # In such case, we must ensure there's another move with the remaining
+        # quantity to do, so selected_line is extracted in a new move as expected.
+
+        # Always keep the quantity todo at zero, the same is done
+        # in Odoo when move lines are created manually (setting)
+        lines_with_qty_todo = selected_line.move_id.move_line_ids.filtered(
+            "product_uom_qty"
         )
-        if new_move:
-            # A new move is created in case of partial quantity
-            new_move.extract_and_action_done()
-            return
-        # In case of full quantity, post the initial move
-        selected_line.move_id.extract_and_action_done()
+        if lines_with_qty_todo:
+            lines_with_qty_todo.product_uom_qty = 0
```

Comment:
We should ensure that there is only one line with product_uom_qty on _scan_line__find_or_create_line we should ensure that the product_uom_qty is 0. Otherwise we can find maybe more then one line?

Reply:
I also improved the filter above to exclude on the state as well, just in case and to be more explicit.

Reply:
What i mean. In _post_line you are setting the product_uom_qty
This line will be found bei auto_post_line normally it should only find the selected_line,
because other lines from other users should also have set  a qty of 0. 
But in _scan_line__find_or_create_line there could be a existing unassigned move.line returned with a qty set.

Also i think we can remove lines_with_qty_todo and always use the selected line.

Reply:
Is it really needed to check for other stock.move.lines with a product_uom_qty > 0?
Because i think every scanned line should have a product_uom_qty of 0, until _post_line is called.

Reply:
yes, because nobody prevents anybody to modify this from the UI.
Hence, we might have issues if `sum(move.move_line_ids.mapped("product_uom_qty")) > move.product_uom_qty`, when recomputing the state of the move.

Reply:
Thx for the explanation. I tested this quick. 
If i'm working with two users. Both are scanning both getting an own move.line. 
The first one creates a new package and sets the the destination and enters the _post_line method.
Within the _auto_post_line the split is happening a new picking is created in state done.
And the move._recompute_state() sets the state to "confirmed". 
So the move state is not anymore 'assigned' as you said. 
What is clear if i am looking at the code. 
https://github.com/odoo/odoo/blob/f572335756ef60402236cf5e4e11c0c843c3879e/addons/stock/models/stock_move.py#L1653
https://github.com/odoo/odoo/blob/f572335756ef60402236cf5e4e11c0c843c3879e/addons/stock/models/stock_move.py#L1659C25-L1659C26
It will only be set to 'assigned' if the move.reserved_availability ( sum of move.lines product_uom_qty) is the same as the move's product_uom_qty, and this isn't the case.

Reply:
I could reproduce the issue you mentioned.
My last commit fixes that.